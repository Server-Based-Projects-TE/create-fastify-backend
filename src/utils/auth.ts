import { RouteShorthandOptions } from 'fastify';
import { createError } from './errors';

export const requireUser: RouteShorthandOptions = {
  preValidation: async (request) => {
    console.dir(request.user);
    if (!request.isAuthenticated()) {
      throw createError(401);
    }
  },
};
