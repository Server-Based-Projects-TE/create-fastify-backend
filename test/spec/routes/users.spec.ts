import { buildTestFastify } from 'test/utils';

describe('app', () => {
  const fastify = buildTestFastify({ bypassAuth: true });
  afterAll(() => {
    fastify.close();
  });
  it('should receive a 200 on GET /users', async () => {
    const response = await fastify.inject({
      method: 'GET',
      url: '/users',
    });
    expect(response.statusCode).toEqual(200);
    expect(response.json()).toMatchSnapshot();
  });
});
