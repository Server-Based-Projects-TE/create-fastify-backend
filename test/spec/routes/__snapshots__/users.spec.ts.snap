// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should receive a 200 on GET /users 1`] = `
Array [
  Object {
    "id": "1",
    "username": "foo",
  },
  Object {
    "id": "2",
    "username": "bar",
  },
  Object {
    "id": "3",
    "username": "baz",
  },
]
`;
